Kyc wip  issue 19 issue 28

kyc-wip opened the pull request:
#19 Add a track to the playlist using a Spotify track id
#28 Add a track to the playlist using a Spotify track name
Functionality exists

TO DO:
1. UX
2. Testing including with Circle CI
3. Render or redirect output to which page

kyc-wip committed:
[#19 #28] Add .py files to call Spotipy methods

Add .py files to TeamGroove folder to call Spotipy methods:
 - which add track (by track_id) to Spotify playlist (by playlist_id)
 - which use Spotify search and returns maximum limit of results (intended for track name)

Requires test and integration

kyc-wip committed:
[#19 #28] Add Scenarios

Add Scenario placeholders .py files in main TeamGroove folder

Requires integration and test.
In preparation:
 - Add urls to main urls.py
 - Add .html files to TeamGroove
 - Add pseudo-random track link to spotify/user_playlists.html

kyc-wip committed:
[#19 #28] Add draft .py file content to spotify/views.py

Add draft add_track_id_to_playlist.py and search_track_name_on_spotify.py file content to spotify/views.py.
Delete draft .py files and associated .html files.

Requires integration, test and refactoring
In preparation:
- Print statements converted to messages
- Add url to handle_search_results.html to main urls.py
- Add handle_search_results.html file to TeamGroove
- Add example test parameters (commented) to spotify/test.py

kyc-wip committed:
Merge branch 'main' into kyc-wip--issue-19-issue-28

kyc-wip committed:
Merge branch 'main' into kyc-wip--issue-19-issue-28

kyc-wip committed:
[#19 #28] Integrate add_track_id_to_playlist and search_track_name_on_spotify

Add tracks to database at user_playlist_tracks
Add track_artist to Track model
Functions add_track_id_to_playlist and search_track_name_on_spotify intended to work with user_playlist_tracks and route to existing urls
Handle search results url removed

Issues:
 - Authentication
 - Some functions can be moved to spotify/utils.py file (example: append track results at user_playlist_tracks function and spotify/utils.py append_track_data perform similar functions and may not need authentication)
 - Requires run and test upon authentication (i.e. no problems show in console but without running cannot be sure)
 - Messages may require formatting

kyc-wip committed:
Merge branch 'snj_fix_env_variables' into kyc-wip--issue-19-issue-28

kyc-wip committed:
[#28]

Add "search_track_name" functionality from room.html and urls.py to spotify folder including new forms.py
Requires test with correct authentication to check flow down to "add_search_result_to_room" and "add_track_id_to_playlist"

Correction to utils.py

Add to spotify.views.py, add_playlist_to_room function:
 - Add Playlist object creation check to prevent repeat playlists in same room, info messages
 - Add Track object creation
Requires test

kyc-wip committed:
[#28] Search track name function

Add search track name function and page
Remove playlist name from URLs

kyc-wip committed:
Merge branch 'main' into kyc-wip--issue-19-issue-28

kyc-wip committed:
[#28] Search results output

Add search results page

TO DO:
- Add search result to playlist
- Code re-factoring and test

kyc-wip committed:
Merge branch 'main' into kyc-wip--issue-19-issue-28

kyc-wip committed:
[#28] Add search results to playlist

Add search results to playlist then return to room

stiffneckjim commented:
I tried running this but when I attempted to add a Spotify playlist to an existing room I got an error.

kyc-wip committed:
Post-test

sfkleach committed:
Remove debug code

sfkleach commented on spotify/utils.py:
```diff
@@ -0,0 +1,23 @@
+            track_id.append(returned_data['tracks']['items'][index]['id'])
+        except IndexError:
+            break
+        else:
+            pass
```
`else: pass` - does nothing

AsquaredWsquared commented on spotify/views.py:
```diff
@@ -74,7 +115,6 @@ def user_playlist_tracks(request, playlist_id):
 
         context = {
             'playlist_id': playlist_id,
-            'playlist_name': playlist_name,
```
This is a workaround but is now fixed elsewhere.

sfkleach commented on spotify/views.py:
```diff
@@ -44,19 +46,58 @@ def authorize_with_spotify(request, spotify_code=None):
+    for track in user_playlist_tracks['items']:
+        list_of_track_ids.append(track['track']['id'])
+
+    results = spotify.tracks(list_of_track_ids)
```
This code has been moved to main.py - we think it is a bad merge

sfkleach commented on spotify/views.py:
```diff
@@ -89,15 +129,132 @@ def add_playlist_to_room(request, playlist_id):
+        messages.error(request, 'ERROR: Track already on playlist')
+        return redirect('room', room_id=room.id) 
 
     return redirect('room', room_id=request.user.active_room_id)
```
This return belong with the `messages.success(request, 'SUCCESS? Add track id to playlist attempted.')`

kyc-wip commented on spotify/utils.py:
```diff
@@ -0,0 +1,23 @@
+    # TRACK_ARTIST, TRACK_NAME and TRACK_ALBUM are intended for view
+    # TRACK_ID is used for adding to a playlist        
+    index = 0        
+    while index < LIMIT:
```
use enumerate and iterate as per for loop in views.py line 76

kyc-wip committed:
Merge branch 'main' into kyc-wip--issue-19-issue-28

kyc-wip committed:
Merge branch 'kyc-wip--issue-19-issue-28' of https://github.com/CodeHubOrg/TeamGroove into kyc-wip--issue-19-issue-28

kyc-wip committed:
Revert "Merge branch 'kyc-wip--issue-19-issue-28' of https://github.com/CodeHubOrg/TeamGroove into kyc-wip--issue-19-issue-28"

This reverts commit ea9f40800e0b78d3235e231d6e51363a6c246dfa, reversing
changes made to 4f2a29fed2a2075dfe098c831deab44485623ade.

kyc-wip committed:
Update Spotify views.py to aligns better with main branch

kyc-wip committed:
Spotify views.py

Bugs fixed up from top of spotify views.py and associated files (forms.py, urls.py, html files)
Debug required from add_track_id_to_playlist function downwards

kyc-wip committed:
Add track id to playlist

Fixes add_track_id_to_playlist function

kyc-wip committed:
Disable and hide playlist_id on search track form

kyc-wip committed:
Enable playlist_id on search track form

Enable playlist_id on search track form otherwise HttpResponse Error
Increase number of search results from 2 sets of 2 (i.e 4), to 2 sets of 10 (i.e. 20)

Requires:
- UX (HTML and CSS) considerations
-  Function to delete track from playlist, db and Spotify playlist
